apps/dsaparam.c

```diff
@@ -180,24 +180,27 @@ int dsaparam_main(int argc, char **argv)
         data = app_malloc(len + 20, "BN space");
 
         BIO_printf(bio_out, "DSA *get_dsa%d()\n{\n", bits_p);
-        print_bignum_var(bio_out, p, "dsap", len, data);
-        print_bignum_var(bio_out, q, "dsaq", len, data);
-        print_bignum_var(bio_out, g, "dsag", len, data);
+        print_bignum_var(bio_out, p, "dsap", bits_p, data);
+        print_bignum_var(bio_out, q, "dsaq", bits_p, data);
+        print_bignum_var(bio_out, g, "dsag", bits_p, data);
         BIO_printf(bio_out, "    DSA *dsa = DSA_new();\n"
+                            "    BIGNUM *p = BN_new(), *q = BN_new(), *g = BN_new();\n"
```

Comment:
You might want to check p, q and g for NULL too.

Reply:
DSA_set0_pqg does it.

Reply:
Though if NULL is passed to BN_bin2bn it will try to allocate, and if succeeds you won't have control over it. Yeah, it would be appropriate to test for NULL...

Reply:
Fixed.